testing/integration/install_test.go

```diff
@@ -0,0 +1,140 @@
+package integration
+
+import (
+	"context"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestInstall(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		OS: []define.OS{
+			define.OS{Type: "darwin"},
+			define.OS{Type: "linux"},
+			define.OS{Type: "windows"},
+		},
+
+		// We require sudo for this test to run
+		// `elastic-agent install`.
+		Sudo: true,
+
+		// It's not safe to run this test locally as it
+		// installs Elastic Agent.
+		Local: false,
+	})
+
+	// Get path to Elastic Agent executable
+	fixture := define.NewFixture(t)
+
+	suite.Run(t, newInstallTestSuite(fixture))
+}
+
+func newInstallTestSuite(fixture *atesting.Fixture) *InstallTestSuite {
+	i := new(InstallTestSuite)
+	i.fixture = fixture
+
+	return i
+}
+
+type InstallTestSuite struct {
+	suite.Suite
+	fixture *atesting.Fixture
+}
+
+func (i *InstallTestSuite) TestInstallWithoutBasePath() {
+	// Check that default base path is clean
+	var defaultBasePath string
+	switch runtime.GOOS {
+	case "darwin":
+		defaultBasePath = `/Library`
+	case "linux":
+		defaultBasePath = `/opt`
+	case "windows":
+		defaultBasePath = `C:\Program Files`
+	}
+
+	topPath := filepath.Join(defaultBasePath, "Elastic", "Agent")
+	_, err := os.Stat(topPath)
+	i.Require().True(os.IsNotExist(err))
+
+	// Run `elastic-agent install`.  We use `--force` to prevent interactive
+	// execution.
+	_, err := i.fixture.Exec(context.Background(), []string{"install", "--force"})
+	i.Require().NoError(err)
+
+	// Check that Agent was installed in default base path
+	i.checkInstallSuccess(topPath)
+}
+
+func (i *InstallTestSuite) TestInstallWithBasePath() {
+	const basePathFlag = "--base-path"
+
+	// The `--base-path` flag is defined for the `elastic-agent install` CLI sub-command BUT
+	// it is hidden (see https://github.com/elastic/elastic-agent/pull/2592).  So we validate
+	// here that the usage text for the `install` sub-command does NOT mention the `--base-path`
+	// flag in it.
+	output, err := i.fixture.Exec(context.Background(), []string{"help", "install"})
+	i.Require().NoError(err)
+	require.NotContains(i.T(), string(output), basePathFlag)
+
+	// Set up random temporary directory to serve as base path for Elastic Agent
+	// installation.
+	tmpDir := os.TempDir()
+	randomBasePath := filepath.Join(tmpDir, strings.ToLower(randStr(8)))
+	defer os.RemoveAll(randomBasePath)
+
+	// Run `elastic-agent install`.  We use `--force` to prevent interactive
+	// execution.
+	_, err = i.fixture.Exec(context.Background(), []string{"install", basePathFlag, randomBasePath, "--force"})
+	i.Require().NoError(err)
+
+	// Check that Agent was installed in the custom base path
+	topPath := filepath.Join(randomBasePath, "Elastic", "Agent")
+	i.checkInstallSuccess(topPath)
```

Comment:
You have 2 tests that are installing the Elastic Agent, without uninstalling before the installation of the next one. This will fail.

We might want to add `Install()` to the fixture as a native function, that registers a cleanup method to uninstall the Elastic Agent. We should probably also do some extra checks and ensure that its fully removed, if not error the test.

I have a feeling install an Elastic Agent will be done by a lot of tests, so adding this management of installing and uninstall/cleanup directly in the fixture is probably needed.

Reply:
> You have 2 tests that are installing the Elastic Agent, without uninstalling before the installation of the next one. This will fail.

Ugh, you're right.  I need to add a post-test cleanup, probably via a [`TearDownTest()` method](https://pkg.go.dev/github.com/stretchr/testify/suite#TearDownTestSuite) on the suite.

> We might want to add `Install()` to the fixture as a native function, ...

I'm ++ to adding an `Install()` method to the fixture, and perhaps more generally a method for each sub-command of `elastic-agent` in the future.  

> ... that registers a cleanup method to uninstall the Elastic Agent. We should probably also do some extra checks and ensure that its fully removed, if not error the test.

> I have a feeling install an Elastic Agent will be done by a lot of tests, so adding this management of installing and uninstall/cleanup directly in the fixture is probably needed.

How do you imagine this working, in relation to the test lifecycle? Are you suggesting that the `Install()` method on the fixture would return a cleanup function that the test code could call via a `defer`? Or something else?

Reply:
Can you elaborate a bit on what you're thinking here, specifically on this bit?

> ... that registers a cleanup method to uninstall the Elastic Agent.

I was thinking of keeping it simple and defining an `Uninstall` method on the fixture that the test code could call either via a `defer` or via a [TearDownTest() method](https://pkg.go.dev/github.com/stretchr/testify/suite#TearDownTestSuite) on the suite.

Reply:
Implemented in 90aa194c4.  Let me know what you think!